modules/deriving/src/main/scala/shapeless3/deriving/kinds.scala

```diff
@@ -166,28 +158,27 @@ object K0:
     inline def widen[G[t] >: F[t]]: CoproductInstances[G, T] =
       inst.asInstanceOf
 
-  inline given mkInstances[F[_], T](using gen: Generic[T]): Instances[F, T] =
+  inline given mkInstances[F[_], T](using gen: Mirror.Of[T]): Instances[F, T] =
```

Comment:
`K0.Generic`/`Refl` isn't actually used here anymore, that's why your test compiles for `show` despite the parameter only asking for a `Mirror`, thus lacking the `type Kind` refinement.

Reply:
`type Generic[O] = Mirror.Of[O] { type Kind = K0.type }`
So `K0.Generic` is the same as `Mirror.Of` plus `type Kind` refinement (for implicit scope).
And `Refl` is the equivalent of `Mirror.Of` for higher kinds.